packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/CameraApiImpl.java

```diff
@@ -0,0 +1,480 @@
+  }
+
+  // We move catching CameraAccessException out of onMethodCall because it causes a crash
+  // on plugin registration for sdks incompatible with Camera2 (< 21). We want this plugin to
+  // to be able to compile with <21 sdks for apps that want the camera and support earlier version.
+  @SuppressWarnings("ConstantConditions")
+  private <T> void handleException(Exception exception, Messages.Result<T> result) {
+    if (exception instanceof CameraAccessException) {
+      result.error(exception);
+      return;
+    }
+
+    // CameraAccessException can not be cast to a RuntimeException.
+    throw (RuntimeException) exception;
+  }
+
+  private void handleException(Exception exception, Messages.VoidResult result) {
+    if (exception instanceof CameraAccessException) {
+      result.error(exception);
+      return;
+    }
+
+    // CameraAccessException can not be cast to a RuntimeException.
+    throw (RuntimeException) exception;
+  }
+
+  @NonNull
+  @Override
+  public List<Messages.PlatformCameraDescription> getAvailableCameras() {
+    if (activity == null) {
+      return Collections.emptyList();
+    }
+    try {
+      return CameraUtils.getAvailableCameras(activity);
+    } catch (CameraAccessException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void create(
+      @NonNull String cameraName,
+      @NonNull Messages.PlatformMediaSettings settings,
+      @NonNull Messages.Result<Long> result) {
+    if (camera != null) {
+      camera.close();
+    }
+
+    cameraPermissions.requestPermissions(
+        activity,
+        permissionsRegistry,
+        settings.getEnableAudio(),
+        (String errCode, String errDesc) -> {
+          if (errCode == null) {
+            try {
+              result.success(instantiateCamera(cameraName, settings));
+            } catch (Exception e) {
+              handleException(e, result);
+            }
+          } else {
+            result.error(new Messages.FlutterError(errCode, errDesc, null));
+          }
+        });
+  }
+
+  @Override
+  public void initialize(
+      @NonNull Long cameraId,
+      @NonNull Messages.PlatformImageFormatGroup imageFormat,
+      @NonNull Messages.VoidResult result) {
+    if (camera != null) {
+      try {
+        camera.open(CameraUtils.imageFormatGroupFromPigeon(imageFormat));
+        result.success();
+      } catch (Exception e) {
+        handleException(e, result);
+      }
+    } else {
+      result.error(
+          new Messages.FlutterError(
+              "cameraNotFound",
+              "Camera not found. Please call the 'create' method before calling 'initialize'.",
+              null));
+    }
+  }
+
+  @Override
+  public void takePicture(@NonNull Long cameraId, @NonNull Messages.Result<String> result) {
+    camera.takePicture(result);
+  }
+
+  @Override
+  public void prepareForVideoRecording(@NonNull Messages.VoidResult result) {
+    // This optimization is not required for Android.
```

Comment:
No-ops should be no-op'd in Dart rather than native; there's a lot of overhead to doing it here.